resources/assets/lib/gallery-contest-vote-progress.tsx

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
+// See the LICENCE file in the repository root for full licence text.
+
+import { CircularProgress } from 'circular-progress';
+import * as React from 'react';
+
+type Props = Record<string, never>;
+
+interface State {
+  voteLeft: number;
+}
+
+export default class GalleryContestVoteProgress extends React.PureComponent<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      voteLeft: this.getVoteLeft(),
+    };
+  }
+
+  componentDidMount() {
+    $.subscribe('contest:vote:end.count', this.syncState);
+  }
+
+  componentWillUnmount() {
+    $.unsubscribe('.count');
+  }
+
+  render() {
+    return (
+      <div className='pswp__button pswp__button--vote-progress'>
+        <CircularProgress
+          current={this.state.voteLeft}
+          max={this.getMaxVotes()}
```

Comment:
make it part of state (and computed together in syncState) otherwise might as well just get both votes left and this during render.

And I just noticed max votes is already fetched in getVoteLeft anyway.